Prog/TOPA/TOPA_add_subsidies.sas

```diff
@@ -0,0 +1,169 @@
+	TOPA_nlihc_dates.ID, TOPA_nlihc_dates.u_address_id_ref, TOPA_nlihc_dates.FULLADDRESS, /** Other vars keeping **/
+	TOPA_nlihc_dates.u_sale_date, TOPA_nlihc_dates.u_notice_date, TOPA_nlihc_dates.u_sum_units,
+	Subsidy.POA_end, Subsidy.POA_end_actual, Subsidy.POA_end_prev,
+	Subsidy.POA_start, Subsidy.POA_start_orig, Subsidy.Portfolio, 
+	Subsidy.Program, Subsidy.Units_Assist, Subsidy.rent_to_fmr_description, Subsidy.Subsidy_id
+  from TOPA_nlihc_dates left join PresCat.Subsidy as Subsidy /** Left join = only keep obs that are in TOPA_nlihc_id**/
+  on TOPA_nlihc_dates.Nlihc_id = Subsidy.Nlihc_id
+  order by TOPA_nlihc_dates.ID; 
+quit; 
+
+%File_info( data=TOPA_subsidy_match, printobs=10 )
+
+** Creating all_POA_start, actual_POA_start, and u_days_notice_to_subsidy, filtering out unneeded obs**; 
+
+data TOPA_subsidy; 
+  set TOPA_subsidy_match; 
+  format all_POA_start MMDDYY10.;
+  format actual_POA_start DYESNO.;
+  format date_LEC_form MMDDYY10.;
+  if not(missing(POA_start_orig)) then do;
+	all_POA_start=POA_start_orig;
+	actual_POA_start=1;
+		end;
+  else if (missing(POA_start_orig)) then do; 
+    all_POA_start=POA_start_orig;
+	actual_POA_start=0;
+		end;
+  label all_POA_start='Period of affordability, original start date or if original date missing, period of affordability, current start date';
+  label actual_POA_start='Original start date used';
+  u_days_notice_to_subsidy = all_POA_start - u_sale_date;
+  label u_days_notice_to_subsidy='Number of days from notice of sale to start of subsidy (Urban created var)';
+  if not(missing(all_POA_start)); ** delete missing subsidy start dates **; 
+  if portfolio in ("LIHTC", "202/811", "PB8", "PRAC", "DC HPTF", "LECOOP"); ** only include LIHTC, federal project-based subsidies, DC HPTF, and LEC **;
+  if portfolio = "LIHTC" and u_days_notice_to_subsidy < 0 then before_LIHTC_aff_units=Units_Assist;
+  if portfolio = "LIHTC" and 0 <= u_days_notice_to_subsidy <= 365 then after_LIHTC_aff_units=Units_Assist;
+  if portfolio in ( "202/811", "PB8", "PRAC" ) and u_days_notice_to_subsidy < 0 then before_fed_aff_units=Units_Assist;
+  if portfolio in ( "202/811", "PB8", "PRAC" ) and 0 <= u_days_notice_to_subsidy <= 365 then after_fed_aff_units=Units_Assist;
+  if portfolio = "DC HPTF" and u_days_notice_to_subsidy < 0 then before_DC_HPTF_aff_units=Units_Assist;
+  if portfolio = "DC HPTF" and 0 <= u_days_notice_to_subsidy <= 365 then after_DC_HPTF_aff_units=Units_Assist;
+  if portfolio = "LECOOP" and u_days_notice_to_subsidy < 0 then before_LEC_aff_units=Units_Assist;
+  if portfolio = "LECOOP" and 0 <= u_days_notice_to_subsidy <= 365 then after_LEC_aff_units=Units_Assist;
+  if missing(poa_end_actual) or u_sale_date-poa_end_actual > 365 or poa_end_actual > '01Jan2023'd; **not including units where subsidy's expired **;
+run;
+
+%File_info( data=TOPA_subsidy, printobs=10 )
+
+**Aggregating the data across rows with the same ID **;
+
+proc summary data=TOPA_subsidy
```

Comment:
When using a CLASS statement to specify the variables for aggregation, you usually want to add the NWAY option to the Proc Summary statement. By default, Proc Summary with CLASS creates intermediate totals as well as a grand total, which is not what we want. That's why the first obs in Topa_sum_rows is ID=.. That is the grand total of all rows. NWAY specifies that you only want the breakdowns using all CLASS variables at once, which is what we want here.